plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -495,7 +498,8 @@ public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, Con
         HiveIdentity identity = new HiveIdentity(session);
         String targetPath = getTableLocation(tableMetadata.getProperties());
         if (targetPath == null) {
-            targetPath = getTableDefaultLocation(database, hdfsContext, hdfsEnvironment, schemaName, tableName).toString();
+            String uniqueTableName = useUniqueTableLocation ? generateUniqueTableName(tableName) : tableName;
```

Comment:
The uniqueTableName variable name suggests we're making the table name unique, while we're only changing table location.

Let's call the variable `tableNameForLocation`

Since same naming problem applies to `generateUniqueTableName`, let's inline that method.

```
String tableNameForLocation = tableName;
if (useUniqueTableLocation) {
    tableNameForLocation += "-" + randomUUID().toString().replace("-", "")
}
```

Reply:
@findepi sure, I will rebase this brach, add code fixes and tests in next few days

Reply:
@findepi I have rebased this brach.
Added few tests to check that uuid suffix added to a table location, and also added tests for check that table correctly dropped .
Will be really appreciate for the review